### ISSUE TITLE
Tiny devices

### ISSUE THREAD

**Issue by mcollina** [creationix/dukluv#1]
It seems libuv runs only on unixes and Windows. Are you still aiming at small devices running these operating systems (say OpenWRT), or are you aiming also at embedded devices?

Anyway, I'm in. I see a big potential in this!

**Comment by prust** [comment]
Sorry to comment on an old thread, but have you looked at [mbed TLS](https://tls.mbed.org) ([formerly PolarSSL](http://community.arm.com/groups/internet-of-things/blog/2015/02/09/polarssl-is-dead-long-live-mbed-tls) -- libcurl [supports compiling with it](https://tls.mbed.org/kb/how-to/compile-curl-with-mbedtls)) as a lightweight alternative to OpenSSL and [miniz](https://code.google.com/p/miniz/) ([github](https://github.com/richgel999/miniz)) as a lightweight alternative to zlib?

There are times when it would be nice to have the full node.js API, but there are other times when I would prefer a small, self-contained executable with just the pieces I need (even avoiding libuv when it isn't necessary).

**Comment by creationix** [comment]
Thanks for the links.  I'm actually using miniz for luvit.io (with a few patches I added to fix it's buggy zip file support.)  We're still using openssl, but I think switching to mbed would shrink the binary a bit.  I'll look into doing the same for dukluv when I get back to working on this.

**Comment by prust** [comment]
@creationix: Great! Are the patches included in https://github.com/creationix/dukluv/blob/master/lib/miniz.c ?

**Comment by prust** [comment]
Follow-up: it looks like the CRC patch (https://github.com/luvit/luvi/issues/113) is included in dukluv's copy of miniz.

**Comment by creationix** [comment]
Right, when in doubt, look at luvit's miniz code, I try to copy all patches here, but sometimes forget.
